### PR TITLE
feat(#2399): new syntax ebnf

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #2399 
Closes: #2407

---

## PR-Codex overview
This PR focuses on various changes in the codebase, including deletion of files, modifications in YAML files, changes in Java code, and updates to the pom.xml file.

### Detailed summary
- Deletion of several YAML files in the `eo-parser` module
- Modifications in YAML files in the `eo-parser` module
- Changes in Java code in the `eo-maven-plugin` and `eo-parser` modules
- Updates to the `pom.xml` file in the `eo-parser` module

> The following files were skipped due to too many changes: `eo-parser/src/test/java/org/eolang/parser/XMIRTest.java`, `eo-runtime/src/test/eo/org/eolang/bytes-tests.eo`, `eo-parser/src/main/antlr4/org/eolang/parser/Program.g4`, `eo-parser/src/main/java/org/eolang/parser/XeListener.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java [hunk 1]
```diff
@@ -88,11 +88,11 @@ void pullsSuccessfully(@TempDir final Path temp) throws IOException {
     void pullsFromProbes(@TempDir final Path temp) throws IOException {
         new FakeMaven(temp)
             .withProgram(
-                "+package org.eolang.custom",
-                "",
```
Comment (yegor256) [inline eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:92]: @maxonfjvipon this empty line is forbidden now? Why did you remove it?

Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:92]: @yegor256 empty line is mandatory there, I just replaced it with `\n` in the code

File: eo-parser/src/test/java/org/eolang/parser/XMIRTest.java [hunk 2]
```diff
@@ -77,6 +78,7 @@ final class XMIRTest {
      * @since 0.30.0
      */
     @Test
+    @Disabled
```
Comment (yegor256) [inline eo-parser/src/test/java/org/eolang/parser/XMIRTest.java:81]: @maxonfjvipon same here. What happened?

**Comment by yegor256** [comment]
@Graur please, take a look too

**Comment by yegor256** [comment]
@maxonfjvipon looks good for me! Great work! Please, see some comments from me above.

File: eo-maven-plugin/src/test/java/org/eolang/maven/SafeMojoTest.java [hunk 3]
```diff
@@ -45,14 +45,14 @@ void logsStackTrace(final Logs out, @TempDir final Path temp) {
         Assertions.assertThrows(
             IllegalStateException.class,
             () -> new FakeMaven(temp)
-                .withProgram("something < is definitely wrong here")
+                .withProgram("something > is definitely wrong here")
```
[the remark this answers is not in the document]
Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/SafeMojoTest.java:48]: @yegor256 this is about `<` (vertex) object. Should it be possible to use it without dot notation? Just like `$` or `&`. For example:
```
[] > int
  [x] > eq
    eq.
      < # <--- is this valid?
      x.<
```

File: eo-parser/src/test/java/org/eolang/parser/XMIRTest.java (hunk 2 above)
Comment (maxonfjvipon) [reply eo-parser/src/test/java/org/eolang/parser/XMIRTest.java:81]: @yegor256 for some reason it [failed](https://github.com/objectionary/eo/actions/runs/6085373156/job/16509280959). I think I should add todo here too

File: eo-maven-plugin/src/test/java/org/eolang/maven/SafeMojoTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/SafeMojoTest.java:48]: @yegor256 our typical usages of `<` are: `object.<`, `^.<`, `$.<`. Everywhere we `.<` (dot vertex), but can `<` be in the head of the sequence without any object before it?

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look one more time

File: eo-maven-plugin/src/test/java/org/eolang/maven/SafeMojoTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (yegor256) [reply eo-maven-plugin/src/test/java/org/eolang/maven/SafeMojoTest.java:48]: @maxonfjvipon got it, you are right

File: eo-parser/pom.xml [hunk 4]
```diff
@@ -188,6 +188,15 @@ SOFTWARE.
       <id>qulice</id>
       <build>
         <plugins>
+          <plugin>
+            <groupId>com.qulice</groupId>
+            <artifactId>qulice-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>checkstyle:/src/main/java/org/eolang/parser/XeListener.java</exclude>
```
Comment (yegor256) [inline eo-parser/pom.xml:196]: @maxonfjvipon this is pretty dangerous. I suggest you add `todo` here.

**Comment by yegor256** [comment]
@Graur I'm ready to mere, what do you say?

**Comment by Graur** [comment]
@yegor256 I think we can try to merge it

**Pull request merged**